src/drake/shell.clj

```diff
@@ -103,7 +103,7 @@
         proc (.exec (Runtime/getRuntime)
                     cmd-for-exec
                     ^"[Ljava.lang.String;" env
-                    (fs/file fs/*cwd*))]
+                    (fs/file (or exec-dir fs/*cwd*)))]
```

Comment:
Very interestingly, when I purposely throw an exception the output makes sense, and in fact shows a different (non-nil) passed in exec-dir. Race condition fun time.

```clojure
        _ (println "exec dir" exec-dir)
        _ (println "cwd" fs/*cwd*)
        _ (println "or'd" (or exec-dir fs/*cwd*))
        _ (println "cwd path" (fs/file fs/*cwd*))
        _ (println "or'd path" (fs/file (or exec-dir fs/*cwd*)))
        dir (fs/file (or exec-dir fs/*cwd*))
        _ (throw (Exception.))
        proc (.exec (Runtime/getRuntime)
                    cmd-for-exec
                    ^"[Ljava.lang.String;" env
                    dir)]
```

```
exec dir /home/chen/code/drake/sandbox/inner
cwd #<File /home/chen/code/drake/sandbox>
or'd /home/chen/code/drake/sandbox/inner
cwd path #<File /home/chen/code/drake/sandbox>
or'd path #<File /home/chen/code/drake/sandbox/inner>
java.lang.Exception
```